Zend/zend_exceptions.c

```diff
@@ -953,16 +978,20 @@ static void zend_error_helper(int type, const char *filename, const uint lineno,
 	va_end(va);
 }
 
-/* This function doesn't return if it uses E_ERROR */
-ZEND_API void zend_exception_error(zend_object *ex, int severity) /* {{{ */
+/* This function doesn't return if it was handled with E_ERROR */
+ZEND_API int zend_exception_error(zend_object *ex, int severity) /* {{{ */
 {
 	zval exception, rv;
 	zend_class_entry *ce_exception;
+	int retval = FAILURE;
 
 	ZVAL_OBJ(&exception, ex);
 	ce_exception = Z_OBJCE(exception);
 	EG(exception) = NULL;
-	if (ce_exception == parse_error_ce) {
+	if (!instanceof_function(ce_exception, zend_ce_throwable)) {
+		/* technical exceptions, ignore them */
+		retval = SUCCESS;
```

Comment:
This if chain already has an `else` condition which I think covers this case. Maybe change that place instead? (Or remove it, having it two times is confusing).